Docs: Backend plugins conceptual overview (first iteration) (#24668)
grafana/grafana: docs/sources/developers/plugins/backend/grafana-plugin-sdk-for-go.md

+++
title = "Grafana Plugin SDK for Go"
keywords = ["grafana", "plugins", "backend", "plugin", "backend-plugins", "sdk", "documentation"]
type = "docs"
+++

# Grafana plugin SDK for Go

> The Grafana plugin SDK for Go enables building Grafana backend plugins using [Go](https://golang.org/). The SDK provides a high-level framework with API's, utilities and tooling that absstracts away the details of the [plugin protocol]({{< relref "plugin-protocol.md" >}}) and RPC communication so plugin developers do not need to manage either.

Reviewer: ```suggestion
The Grafana plugin SDK for Go enables building Grafana backend plugins using [Go](https://golang.org/). The SDK provides a high-level framework with APIs, utilities and tooling that abstract away the details of the [plugin protocol]({{< relref "plugin-protocol.md" >}}) and RPC communication so plugin developers do not need to manage either.
```

> The Grafana plugin SDK for Go enables building Grafana backend plugins using [Go](https://golang.org/). The SDK provides a high-level framework with API's, utilities and tooling that absstracts away the details of the [plugin protocol]({{< relref "plugin-protocol.md" >}}) and RPC communication so plugin developers do not need to manage either.

Reviewer 2: ```suggestion
The Grafana plugin SDK for Go enables you to build Grafana backend plugins using [Go](https://golang.org/). The SDK provides a high-level framework with API's, utilities, and tooling that abstracts away the details of the [plugin protocol]({{< relref "plugin-protocol.md" >}}) and RPC communication.
```

> The [github.com/grafana/grafana-plugin-sdk-go](https://pkg.go.dev/mod/github.com/grafana/grafana-plugin-sdk-go?tab=overview) is a Go module that provides a set of [Go packages](https://pkg.go.dev/mod/github.com/grafana/grafana-plugin-sdk-go?tab=packages) that can be imported and used to implement a backend plugin.

Reviewer 2: ```suggestion
The [github.com/grafana/grafana-plugin-sdk-go](https://pkg.go.dev/mod/github.com/grafana/grafana-plugin-sdk-go?tab=overview) is a Go module that provides a set of [Go packages](https://pkg.go.dev/mod/github.com/grafana/grafana-plugin-sdk-go?tab=packages) that can be used to implement a backend plugin.
```

## Versioning

> The SDK is still in development. The [plugin protocol]({{< relref "plugin-protocol.md" >}}) between Grafana and the plugin SDK is versioned separatly and considered stable. However, there might be breaking changes introduced in the SDK. This means that plugins using an older version of the SDK should still work with Grafana, but it might lose out on new features and capabilities introduced in the SDK.

Reviewer: ```suggestion
The SDK is still in development. The [plugin protocol]({{< relref "plugin-protocol.md" >}}) between Grafana and the plugin SDK is versioned separately and considered stable. However, there might be breaking changes introduced in the SDK. This means that plugins using an older version of the SDK should still work with Grafana, but might lose out on new features and capabilities introduced in the SDK.
```

> The SDK is still in development. The [plugin protocol]({{< relref "plugin-protocol.md" >}}) between Grafana and the plugin SDK is versioned separatly and considered stable. However, there might be breaking changes introduced in the SDK. This means that plugins using an older version of the SDK should still work with Grafana, but it might lose out on new features and capabilities introduced in the SDK.

Reviewer 2: ```suggestion
The SDK is still in early development. The [plugin protocol]({{< relref "plugin-protocol.md" >}}) between Grafana and the plugin SDK is versioned separately and is considered stable. However, there might be breaking changes introduced in the SDK. This means that plugins using an older version of the SDK should still work with Grafana, but it might lose out on new features and capabilities that are introduced in the SDK in the future.
```

> ## See also

Reviewer 2: ```suggestion
## See also

```

> - Source code repository: https://github.com/grafana/grafana-plugin-sdk-go

Reviewer 2: ```suggestion
- [Source code](https://github.com/grafana/grafana-plugin-sdk-go)
```

> - Go documentation reference: https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go

Reviewer 2: ```suggestion
- [Go reference documentation](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go)
```
